cmake/modules/extensions.cmake

```diff
@@ -2814,6 +2814,91 @@ function(target_byproducts)
   )
 endfunction()
 
+# Usage:
+#   zephyr_topological_sort(VERTICES <v_0> [<v_1> ...]
+#                           [EDGES <v_i>,<v_j> ...]
+#                           OUTPUT_RESULT <out-variable>
+#                           [OUTPUT_CYCLE <out-variable>])
```

Comment:
Instead of trying to make this completely generic then make it dependent on targets.

`add_dependencies()` in CMake already requires targets as arguments, and so far I only see use-cases in CMake where topological sort is performed on target.

Therefore, using targets as input will make API and calling code cleaner, for example:
```
#   zephyr_topological_sort(TARGETS<t0> [<t1> ...]
#                           [PROPERTY_NAME <name>]
#                           OUTPUT_RESULT <out-variable>
#                           [OUTPUT_CYCLE <out-variable>])
#
# PROPERTY_NAME <name>: Name of property on <targets> which contains a list of dependencies used for sorting.
#                       As example, a CMake library property name could be `LINK_DEPENDS`.
```

Reply:
Addressed here: https://github.com/zephyrproject-rtos/zephyr/pull/57884#discussion_r1206841170